### ISSUE TITLE
config: ignore-article: improve readability

### ISSUE THREAD

**Issue by allanwind** [newsboat/newsboat#1951]
**Newsboat version (copy the output of `newsboat -v` or the first line of `git show`)**: bf806b11e9c05645939f1f38d631ebc90facfa73

Config file (copy from ~/.newsboat/config or ~/.config/newsboat/config):

```
article-sort-order title
articlelist-title-format "%U"
bind-key J next-feed
bind-key K prev-feed
bind-key j next
bind-key k prev
bind-key o open-in-browser-and-mark-read articlelist
confirm-mark-feed-read no
datetime-format "%F %T"
error-log ~/.local/share/newsboat/error.log
feedlist-format "%4i %n %11u %L - %t"
ignore-article "*" "rssurl =~ \"u1\" and title =~ \"r1|r2|r3|...|rN\"" 
...
ignore-mode display
keep-articles-days 365
prepopulate-query-feeds yes
reload-threads 32
run-on-startup reload-all;open
show-read-articles no
show-read-feeds no
unbind-key C
```

Mulitple ignore-articles for "*" work (afaik) but the regular expressions (r1, r2, ... rN) are fairly long and I would kindly request an enhancement to improve readability in particular a line per regex.   No preference on syntax: 1. outer "" spanning support multiple lines (with a. magic whitespace stripping when parsed, or b. just insist on a more verbose expression), 2. support a list of strings ("re1"\n"r2"), or c. with the \ line terminate so I can write something like.  Just don't make the configuration white space significant (like Haskell, Python).  Here is how 1.a.:

```
ignore-article "*" "
  rssurl =~ \"u1\" and title =~ \"
    r1|
    r2|
    r3|
    ...
    rN\"
  or
  ...
"
```
and 1.b would look like (syslog-ng support this and it works really well in practice):

```
ignore-article "*" "
  rssurl =~ \"u1\" and (
     title =~ \"r1\" or
     title =~ \"r2\" or
     title =~ \"r3\" or

     title =~ \"rN\"
  ) or
  ...
"
```

The repetition/verbose form doesn't matter but line format is highly useful to quickly diff for changes.

From my network news days, I often miss a quick way to update ignore-articles, and and I am not sure how to solve that UX issue when ignore-article is a complicated expression that I use (rssurl =~ ... and title =~ or rssurl !~ ... and title =~ ...).  Ansible has a concept of an external [lookup](https://docs.ansible.com/ansible/latest/user_guide/playbooks_lookups.html) that might a good way to get the best of both the power of expressions and the easy of a flat list.  Syntax might look something like this:

```
ignore-article "*" "
  rssurl =~ \"u1\" and title =~ lookup("title.killfile") or
  rssurl =~ \"u2\" and author = lookup("author.killfile") or
  ...
 ```

or with multiple ignore-articles:

ignore-article "*" "rssurl =~ \"u1\" and title =~ lookup(\"title.killfile\")"
ignore-article "*'  "rssurl =~ \"u2\" and author = lookup(\"author.killfile\")"

Finally consider deprecating the feed field.   In practice "*" or exact string match of url is limiting, and it misleads you into thinking that newsboat can't do flexible pattern match when indeed supports it with `rssurl =~ ...`.

**Comment by Minoru** [comment]
Starting with Newbsoat 2.30 (to be released this weekend) it's possible to break long lines into multiple ones using `\`, like in shell scripts.

The other two points listed here still stand.

**Comment by allanwind** [comment]
Awesome. Is the syntax documented anywhere?  This didn't work on
master:

ignore-article "*" title =~ \"(prefix)(\
    abc|\
    def\
)(suffix)
"

If it only works between terms then you have to repeat the prefix 
and suffix for each regex.  Irregardless, it is still an 
improvement.

**Comment by Minoru** [comment]
Yes, the syntax is documented in the configuration chapter: https://github.com/newsboat/newsboat/pull/2229/files#diff-98313a64af2c61154d61154536cfa649b6569eabc8921bcf43ac9dd23be76653

The command didn't work because the last argument to `ignore-article` is not enclosed in double quotes, and the string inside it has an opening quote but not the closing one. The fixed version works (i.e. gets parsed by Newsboat just fine):

```
ignore-article "*" "title =~ \"(prefix)(\
    abc|\
    def\
)(suffix)\
\""
```

**Comment by allanwind** [comment]
Sorry, I had the right format just carelessly in the example.  It 
looks like it just isn't merged into mater yet.  Looking forward 
to giving it a whirl when it lands.

**Comment by allanwind** [comment]
Sorry, I was careless when I wrote up the sample config and was actually testing with one of my current rule which failed.   I can't, for the life of me, figure out what I did wrong, as I did `git pull upstream master` followed by  `make install prefix=~/stow/newsboat` but works beautifully now.  Doesn't github usually stick a comment into the ticket for the commit/pr?

**Comment by Minoru** [comment]
Sorry it took me, um, 9 months to respond. I'm not ignoring you (or anyone else), just didn't have the energy to work through all the issues and comments that come in. Anyway:

> if I prefix a line with space or tab I think they up in the regex

Correct. The backslash "swallows" the following newline, as if it wasn't threre — but that's it. Whatever follows the newline ends up in the resulting string. In your case, that's the extra tabs/spaces at the start of the next line.

I understand the usefulness of indentation, but filter expressions aren't a real programming language (at least not yet :) 

Either way, I attempted to improve documentation on this point: https://github.com/newsboat/newsboat/pull/2563 What do you think?
